src/clj/puppetlabs/http/client/async.clj

```diff
@@ -173,11 +173,12 @@
     metric-registry :- (schema/maybe MetricRegistry)
     metric-namespace :- (schema/maybe schema/Str)]
    (let [result (promise)
-         defaults {:headers {}
-                   :body nil
+         defaults {:body nil
                    :decompress-body true
                    :as :stream}
-         opts (merge defaults opts)
+         headers (merge {"Accept-Language" (str (i18n/user-locale))} (:headers opts))
```

Comment:
You should call `java.lang.Locale#toLanguageTag` rather than `#toString` for this. HTTP 1.1 says that the [language tag](https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.10) needs to comply with [RFC 1766](), which as far as I can tell has been superseded by [IETF BCP 47](https://en.wikipedia.org/wiki/IETF_language_tag) as far as I can tell